app/components/Sidebar/components/DocumentLink.js

```diff
@@ -25,139 +23,150 @@ type Props = {|
+          label={
+            <>
+              {hasChildDocuments && (
+                <Disclosure
+                  expanded={expanded}
+                  onClick={handleDisclosureClick}
+                />
+              )}
               <EditableTitle
-                title={title}
-                onSubmit={this.handleTitleChange}
+                title={node.title || "Untitled"}
+                onSubmit={handleTitleChange}
                 canUpdate={canUpdate}
               />
-            }
-            depth={depth}
-            exact={false}
-            menuOpen={this.menuOpen}
-            menu={
-              document ? (
-                <Fade>
-                  <DocumentMenu
-                    position="right"
-                    document={document}
-                    onOpen={() => (this.menuOpen = true)}
-                    onClose={() => (this.menuOpen = false)}
-                  />
-                </Fade>
-              ) : undefined
-            }
-          >
-            {this.hasChildDocuments() && (
-              <DocumentChildren column>
-                {node.children.map((childNode) => (
-                  <DocumentLink
-                    key={childNode.id}
-                    collection={collection}
-                    node={childNode}
-                    documents={documents}
-                    activeDocument={activeDocument}
-                    prefetchDocument={prefetchDocument}
-                    depth={depth + 1}
-                    canUpdate={canUpdate}
-                  />
-                ))}
-              </DocumentChildren>
-            )}
-          </SidebarLink>
-        </DropToImport>
-      </Flex>
-    );
-  }
+            </>
+          }
+          depth={depth}
+          exact={false}
+          menuOpen={menuOpen}
+          menu={
+            document ? (
+              <Fade>
+                <DocumentMenu
+                  position="right"
+                  document={document}
+                  onOpen={() => setMenuOpen(true)}
+                  onClose={() => setMenuOpen(false)}
+                />
+              </Fade>
+            ) : undefined
+          }
+        ></SidebarLink>
+      </DropToImport>
+
+      {expanded && (
+        <>
+          {node.children.map((childNode) => (
+            <ObservedDocumentLink
+              key={childNode.id}
+              collection={collection}
+              node={childNode}
+              activeDocument={activeDocument}
+              prefetchDocument={prefetchDocument}
+              depth={depth + 1}
+              canUpdate={canUpdate}
+            />
+          ))}
+        </>
+      )}
+    </React.Fragment>
+  );
 }
 
-const DocumentChildren = styled(Flex)``;
+const Disclosure = styled(CollapsedIcon)`
+  position: absolute;
+  left: -24px;
+
+  ${({ expanded }) => !expanded && "transform: rotate(-90deg);"};
+`;
 
-export default DocumentLink;
+const ObservedDocumentLink = observer(DocumentLink);
```

Comment:
🍗 the recursive call needs to be wrapped with observer; this seems like the least mysterious way to do it